src/mpc/key_exchange.md

```diff
@@ -1,40 +1,63 @@
 # Key Exchange
 
-In TLS, the first step towards obtaining TLS session keys is to compute a shared secret between the client and the server by running the [ECDH protocol](https://en.wikipedia.org/wiki/Elliptic-curve_Diffie–Hellman). The resulting shared secret in TLS terms is called the pre-master secret `PMS`.
+In TLS, the first step towards obtaining TLS session keys is to compute a shared secret between the client and the server by running the [ECDH protocol](https://en.wikipedia.org/wiki/Elliptic-curve_Diffie–Hellman). The resulting shared secret in TLS terms is called the pre-master secret.
 
-<img src="../../png-diagrams/key_exchange.png" width="800">
+<div class="aligncenter"><img src="../../png-diagrams/key_exchange.png" width="800"></div>
 
-Using the notation from Wikipedia, below is the 3-party ECDH protocol between the `Server` the `Requester` and the `Notary`, enabling the `Requester` and the `Notary` to arrive at shares of `PMS`.
+## ECDH
 
+We will denote:
 
-1. `Server` sends its public key $Q_b$ to `Requester`, and `Requester` forwards it to `Notary`
-2. `Requester` picks a random private key share $d_c$ and computes a public key share $Q_c = d_c * G$
-3. `Notary` picks a random private key share $d_n$ and computes a public key share $Q_n = d_n * G$
-4. `Notary` sends $Q_n$ to `Requester` who computes $Q_a = Q_c + Q_n $ and sends $Q_a$ to `Server`
-5. `Requester` computes an EC point $(x_p, y_p) = d_c * Q_b$
-6. `Notary` computes an EC point $(x_q, y_q) = d_n * Q_b$
-7. Addition of points $(x_p, y_p)$ and $(x_q, y_q)$ results in the coordinate $x_r$, which is `PMS`. (The coordinate $y_r$ is not used in TLS)
+- $P$ as the `Prover`
+- $V$ as the `Verifier`
+- $S$ as the `Server`
+- $\mathbb{pms}$ as the TLS pre-master secret.
 
+Below is the 3-party ECDH protocol between $S$, $P$ and $V$, enabling $P$ and $V$ to arrive at shares of $\mathbb{pms}$.
```

Comment:
Can we mention that the notation is from Wikipedia, so ppl could have wikipedia open and easier follow the explanation.